src/osd/PGBackend.cc

```diff
@@ -832,11 +832,11 @@ map<pg_shard_t, ScrubMap *>::const_iterator
     // This is automatically corrected in PG::_repair_oinfo_oid()
     assert(oi.soid == obj);
 
-    if (auth_version != eversion_t()) {
-      if (!object_error.has_object_info_inconsistency() && !(bl == auth_bl)) {
-	object_error.set_object_info_inconsistency();
-	error_string += " object_info_inconsistency";
-      }
+    if (first_bl.length() == 0) {
+      first_bl.append(bl);
+    } else if (!object_error.has_object_info_inconsistency() && !(bl == first_bl)) {
```

Comment:
@tchaikov This test gets the first one using auth_version == eversion_t() then compares the each subsequent version against the best version so far using the snippet you showed.